yjit/src/asm/arm64/arg/bitmask_imm.rs

```diff
@@ -72,13 +72,31 @@ impl TryFrom<u64> for BitmaskImmediate {
     }
 }
 
-impl From<BitmaskImmediate> for u32 {
+impl BitmaskImmediate {
+    /// Attempt to make a BitmaskImmediate for a 32 bit register.
+    /// The result has N==0, which is required for some 32-bit instructions.
+    /// Note that the exact same BitmaskImmediate produces different values
+    /// depending on the size of the target register.
+    pub fn new_32b_reg(value: u32) -> Result<Self, ()> {
```

Comment:
I opted for an explicit function instead of implementing `TryFrom<u32>` for this one. 1 has a different encoding depending on whether we're encoding for a 32 bit or 64 bit instruction. Making that determination based on the type feels like smuggling information in because if both `TryFrom<u32>` and `TryFrom<u64>` are present, I would expect each numeric value to have a unique encoding, which is not the case here.